ddtrace/opentelemetry/tracer.go

```diff
@@ -44,28 +46,48 @@ func (t *oteltracer) Start(ctx context.Context, spanName string, opts ...oteltra
 	if t := ssConfig.Timestamp(); !t.IsZero() {
 		ddopts = append(ddopts, tracer.StartTime(ssConfig.Timestamp()))
 	}
-	for _, attr := range ssConfig.Attributes() {
-		ddopts = append(ddopts, tracer.Tag(toSpecialAttributes(string(attr.Key), attr.Value)))
-	}
 	if k := ssConfig.SpanKind(); k != 0 {
 		ddopts = append(ddopts, tracer.Tag(ext.SpanKind, k.String()))
 	}
-	if opts, ok := spanOptionsFromContext(ctx); ok {
-		ddopts = append(ddopts, opts...)
-	}
 	telemetry.GlobalClient.Count(telemetry.NamespaceTracers, "spans_created", 1.0, telemetryTags, true)
+	ctxOpts, ok := spanOptionsFromContext(ctx)
+	if ok {
+		ddopts = append(ddopts, ctxOpts...)
+	}
+	var cfg ddtrace.StartSpanConfig
+	for _, option := range ctxOpts {
+		option(&cfg)
+	}
```

Comment:
I'm curious why we need to call these option functions at all. If we add them to the `ddopts` won't the functions get called later anyway? We don't call the options given to us from `spanOptionsFromContext` for example, is why I ask.

Reply:
They will get called indeed, but we will have no idea what was set via these options, and so we will not know what kind of attributes are set, and creating operation name will be incorrect in some cases